### ISSUE TITLE
Cannot get the right widget name in some cases

### ISSUE THREAD

**Issue by lscrd** [StefanSalewski/gintro#22]
I think this problem is not limited to the name.
When I change the name of a Button (or of any other widget except Widget), I can get the new name if I access directly from the button, but if I use a procedure with a Widget as argument, I get the old name.

Here is an example:

```Nim
import gintro/[gtk, gobject, glib]
import gintro/gio except Application, newApplication

proc displayName(widget: Widget) =
  echo widget.name

proc activate(app: Application) =
  let window = newApplicationWindow(app)
  let button = newButton("test")
  button.setName("MyButton")
  echo button.name
  button.displayName()
  window.add(button)
  window.showAll()

proc main() =
  var app = newApplication()
  discard app.connect("activate", test_set_name.activate)
  discard app.run()

main()
```

**Comment by StefanSalewski** [comment]
Fixed. But there may exist more name conflicts...

Note that gio.Application was renamed to GApplication, so

```nim
import gintro/[gtk, gobject, glib, gio]
```

is sufficient. Still have to fix tutorial.

**Comment by lscrd** [comment]
Thanks! It works now and another problem has been solved: I use the widget name in a CSS to change the widget background color and even with the right name it no longer worked. I don’t know why it works again. I didn’t create an issue for this as I had no clue about the cause (it could have been some obscure error in my program).

**Comment by StefanSalewski** [comment]
Fine that it works for you now.

Setting colors with CSS can be difficult indeed, I had some problems in Ruby and C myself, and some other people also. For example I had code that was working before GTK 3.20, but needed fixes for GTK 3.20 and higher. And I have seem reports on StackOverflow and similar pages from people having a lot of trouble setting colors for some widgets and some Themes. Let me know if you have problems again...
